### PR TITLE
Fix: logout and auto-recconect using SessionStorage

### PR DESCRIPTION
**Pull request by BjrInt**

### CODE REVIEW

File: src/components/common/Main/cmp.tsx [hunk 1]
```diff
@@ -1,17 +1,23 @@
 import { useEffect } from 'react'
 import { MainProps } from './types'
 import { useConnect } from '@/hooks/common/useConnect'
+import { useSessionStorage } from 'usehooks-ts'
 
 export const Main = ({ children }: MainProps) => {
   const { isConnected, connect } = useConnect()
+  const [keepAccountAlive] = useSessionStorage(
```
Comment (amalcaraz) [inline src/components/common/Main/cmp.tsx:8]: I would move all this logic inside the useConnect hook, just exporting `tryRecconect` method that will handle this internally,

**Review by amalcaraz — CHANGES_REQUESTED** [review]

File: src/components/common/Main/cmp.tsx (hunk 1 above)
Comment (BjrInt) [reply src/components/common/Main/cmp.tsx:8]: Yeah having this logic dedupped accross two files did not make a lot of sense to me either. 
Changing that.

File: src/hooks/common/useConnect.ts [hunk 2]
```diff
@@ -59,16 +66,23 @@ export function useConnect(): UseConnectReturn {
...
 
   const disconnect = useCallback(async () => {
+    setKeepAccountAlive(false)
     dispatch({ type: ActionTypes.disconnect, payload: null })
   }, [dispatch])
 
   const { account } = state
   const isConnected = !!account?.address
 
+  const tryReconnect = async () => {
```
Comment (amalcaraz) [inline src/hooks/common/useConnect.ts:76]: use `useCallback` here please

File: src/components/common/Main/cmp.tsx [hunk 3]
```diff
@@ -3,14 +3,9 @@ import { MainProps } from './types'
...
+  const { isConnected, connect, tryReconnect } = useConnect()
 
   useEffect(() => {
-    async function tryReconnect() {
-      if (isConnected) return
-      await connect()
-    }
-
     tryReconnect()
   }, [connect, isConnected])
```
Comment (amalcaraz) [inline src/components/common/Main/cmp.tsx:10]: update the list of deps here with just `tryReconnect`. Your IDE should warn you about this, isn't it?

**Review by amalcaraz — CHANGES_REQUESTED** [review]

**Pull request merged**